File: src/leetcode/java/Q19/ListNode.java
package Q19;

/**
 * Created by hao on 6/18/17.
 */
public class ListNode {
    int val;
    ListNode next;
    ListNode(int x) { val = x; }

    public String toString() {
        String res = "";
        ListNode n = this;
        while (n != null) {
            res += n.val;
            if (n.next != null) res += "->";
            n = n.next;
        }
        return res;
    }

    public static ListNode createFromArray(int[] arr) {
        if (arr.length == 0) return null;

        ListNode head = new ListNode(arr[0]);
        ListNode current = head;

        for (int i = 1; i < arr.length; i++) {
            ListNode node = new ListNode(arr[i]);
            current.next = node;
            current = node;
        }

        return head;
    }
}

File: leetcode/javascript/util/ListNode.js
class ListNode {
  constructor(val) {
    this.val = val;
    this.next = null;
  }
}

function arrayToListNode(array) {
  let head;
  let node = null;

  for (let i = 0; i < array.length; i++) {
    let n = new ListNode(array[i]);
    if (i === 0) head = n;
    if (node !== null) node.next = n;
    node = n;
  }
  return head;
}

function listNodeToArray(listNode) {
  let array = [];
  let n = listNode;
  while (n !== null) {
    array.push(n.val);
    n = n.next;
  }
  return array;
}

exports.ListNode = ListNode;
exports.arrayToListNode = arrayToListNode;
exports.listNodeToArray = listNodeToArray;